jina/types/sets/cache.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional
+from .document import DocumentSet
+
+
+class CacheDocumentSet:
+
+    def __init__(self,
+                 capacity: Optional[int] = None,
+                 *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.capacity = capacity
+        self._doc_set = DocumentSet()
+
+    @property
+    def left_capacity(self):
```

Comment:
nit: `left_capacity` could be renamed to `available_capacity` as left indicates subtraction